src/main/java/org/jabref/logic/cleanup/RenamePdfCleanup.java

```diff
@@ -125,22 +122,12 @@ public RenamePdfCleanup(boolean onlyRelativePaths, BibDatabaseContext databaseCo
                 changed = true;
 
                 //Change the path for this entry
-                String description = flEntry.getDescription();
-                String type = flEntry.getFileType();
+                String description = oldLinkedFile.getDescription();
+                String type = oldLinkedFile.getFileType();
 
                 //We use the file directory (if none is set - then bib file) to create relative file links
                 Optional<Path> settingsDir = databaseContext.getFirstExistingFileDir(fileDirectoryPreferences);
-                if (settingsDir.isPresent()) {
-
-                    Path parent = settingsDir.get();
-                    String newFileEntryFileName;
-                    if (parent == null) {
-                        newFileEntryFileName = targetFileName;
-                    } else {
-                        newFileEntryFileName = parent.relativize(newPath).toString();
-                    }
-                    newFileList.add(new LinkedFile(description, newFileEntryFileName, type));
-                }
+                settingsDir.ifPresent(path -> newFileList.add(new LinkedFile(description, path.relativize(newPath).toString(), type)));
```

Comment:
shouldn't be here an `orElse` to ensure that  in `newFileList` the entry is always added, even if there is no `getFirstExistingFileDir`?

Maybe, we can be sure that this always returns something in this state. If we are, we have to add

    if (!settingsdir.isPresent) LOGGER.debug("no directory determined. This is strange.");

Reply:
an orElse would be the wrong approach here because it would then resolve to the JabRef install directory.

Reply:
In fact I think the optional could be dropped because the bib file dir should always be a valid directory